tools/doc/html.js

```diff
@@ -165,10 +175,39 @@ function parseLists(input) {
 }
 
 
+// Gets a link to a Linux or BSD manpage.
+// For example, getManLink("open(2)") returns
+// '<a href="http://man7.org/linux/man-pages/man2/open.2.html">open(2)</a>'
+function getManLink(text) {
+  // It's safe to concatenate into HTML like this because `text` matches
+  var found = text.match(/([a-z]+)\(([0-9])\)/);
+  if (!found) {
+    throw new Error('Expected a man page name in form foo(8)');
+  }
+  var name = found[1];
+  var number = found[2];
+  var displayAs = name + '(' + number + ')';
+  if (name !== 'lchmod') {
+    return ' <a href="http://man7.org/linux/man-pages/man' + number + '/' +
+      name + '.' + number + '.html">' + displayAs + '</a>';
+  } else {
+    return ' <a href="http://www.freebsd.org/cgi/man.cgi?query=' + name +
+      '&sektion=' + number + '">' + displayAs + '</a>';
+  }
+}
+
+// Handle references to man pages, eg "open(2)" or "lchmod(2)"
+// Returns modified text, with such refs replace with HTML links
+function linkManPages(text) {
+  return text.replace(/ [a-z]+\([0-9]\)/, getManLink);
```

Comment:
yeah good call, i changed it to `replaceAll`

i did the grouping there to make it clear that `name` and `number` are safe to concat into an HTML string, but i'll just combine the two methods to keep things simple